source/agora/consensus/state/ValidatorSet.d

```diff
@@ -387,6 +387,34 @@ public class ValidatorSet
         }
     }
 
+    /***************************************************************************
+
+        Get the utxo_key used for enrollemnt for the given public key
+
+        Params:
+            public_key = public key of the validator
+
+        Returns:
+            Return UTXO key as Hash
+
+    ***************************************************************************/
+
+    public Hash getEnrollmentForKey (in PublicKey public_key) @trusted nothrow
+    {
+        try
+        {
+            auto results = this.db.execute("SELECT key FROM validator_set " ~
+                "WHERE public_key = ? ORDER BY active DESC", public_key);
+            if (!results.empty && results.oneValue!(byte[]).length != 0)
+                return Hash(results.front.peek!(char[])(0));
+        }
+        catch (Exception ex)
+        {
+            log.error("ManagedDatabase operation error: {}", ex.msg);
+        }
+        return Hash.init;
+    }
+
```

Comment:
We really ought to return all the infos about a validator instead of exposing one method per column in a row

Reply:
I would prefer if we could have a new issue to address improving the database queries for validator information needed each block.

Reply:
Yes, that's sensible.